### PR TITLE
todoMVC: Migrate to @types and integrate dojo-interfaces.

### LINKED ISSUE #111: todoMVC: Migrate to @types and integrate dojo-interfaces.
See https://github.com/dojo/meta/issues/62 for details.

### PR DESCRIPTION
**Pull request by lzhoucs**
* Converts to `@types` and integrates `dojo-interfaces` into the package;
* Fix additional build issues

Fixes #111

### CODE REVIEW

**Comment by lzhoucs** [comment]
@kitsonk  I run into another issue with this migration. The issue is that an additional wrapper is generated by `tsc` that causes TodoMVC packages defined in `index.ts` not able to load. The following is a snippet of the additional wrapper added in 'todo-mvc/_build/src/index.js':
```javascript
(function (factory) {
    if (typeof module === 'object' && typeof module.exports === 'object') { /*... */ }
    else if (typeof define === 'function' && define.amd) { /*... */ }
})(function (require, exports) {
    require.config({
        baseUrl: '../../',
        packages: [
            { name: 'src', location: '_build/src' }
        ]
    });
    require(['src/main'], function () { });
});
```
Whereas previously, this file looks like below:
```javascript
require.config({
    baseUrl: '../../',
    packages: [
        { name: 'src', location: '_build/src' }
    ]
});
require(['src/main'], function () { });
```

I can't explain why this happened, since we are only changing the typings, and the definition for `RootRequire` that is used in [index.ts](https://github.com/dojo/examples/pull/112/files#diff-34efadcc32df29261afbe6b5f5dd2c6cR1) as shown below should be the same between previous `dojo-loader` and current `dojo-interface`
```typescript
import { RootRequire } from 'dojo-interfaces/loader';
declare const require: RootRequire;
```
The only difference is that we need to explicitly import it since the newly added `@types/node` has a block-scoped variable `require`, and the previous block-scoped variable `require` that existed in `dojo-loader` has been removed from `dojo-interfaces`

**Comment by kitsonk** [comment]
@lzhoucs by importing something, you are changing it into a module, therefore it is getting exported as a module.  The best thing to do in this situation is just `(<any> require).config()`.

**Comment by lzhoucs** [comment]
Thanks Kit. This issue is resolved after switching to `any`.

File: todo-mvc/package.json [hunk 1]
```diff
@@ -15,10 +16,24 @@
     "node": ">=6",
     "npm": ">=3"
   },
+  "peerDependencies": {
```
Comment (Tomdye) [inline todo-mvc/package.json]: @lzhoucs these should have stayed as dependencies. As they are actual dependencies of the TODO app

File: todo-mvc/package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "description": "TodoMVC using Dojo2",
   "main": "index.js",
   "scripts": {
+    "prepublish": "grunt peerDepInstall",
```
Comment (Tomdye) [inline todo-mvc/package.json]: won't need this

File: todo-mvc/package.json (hunk 1 above)
Comment (lzhoucs) [inline todo-mvc/package.json]: Yeah, that makes sense to me. I've fixed it.

**Pull request merged**